### PR TITLE
Create full orders

### PR DESCRIPTION
**Pull request by brendandeere**
A Consolidated installment can produce a Spree::Order with the correct
contents and move it entirely through the checkout process until it is
completed

### CODE REVIEW

**Comment by brendandeere** [comment]
https://www.pivotaltracker.com/story/show/129392957

**Comment by brendandeere** [comment]
This pr depends on #13

File: app/models/solidus_subscriptions/consolidated_installment.rb [hunk 1]
```diff
@@ -42,5 +60,25 @@ def populate
...
+    def subscription
+      installments.first.subscription
+    end
+
+    def ship_address
+      user.ship_address || root_order.ship_address
+    end
+
+    def active_card
+      user.credit_cards.last || root_order.credit_cards.last
```
Comment (brendandeere) [inline app/models/solidus_subscriptions/consolidated_installment.rb]: WIll make this the default card instead

File: app/models/solidus_subscriptions/consolidated_installment.rb [hunk 2]
```diff
@@ -42,5 +60,25 @@ def populate
...
+    end
+
+    def create_payment
+      order.payments.create(
+        source: active_card,
+        amount: order.total,
+        payment_method: Spree::Gateway.where(active: true).first
```
Comment (brendandeere) [inline app/models/solidus_subscriptions/consolidated_installment.rb]: Will make this a config option with the default being the first active gateway with CreditCard as the source class

**Comment by brendandeere** [comment]
Now also depends on #17

File: app/models/solidus_subscriptions/consolidated_installment.rb [hunk 3]
```diff
@@ -21,7 +23,62 @@ def initialize(installments)
...
+    end
+
+    # The order fulfilling the consolidated installment
+    #
+    # @return [Spree::Order]
+    def order
+      @order ||= Spree::Order.create(
+        user: user,
+        email: user.email,
+        store: Spree::Store.default
```
Comment (Sinetheta) [inline app/models/solidus_subscriptions/consolidated_installment.rb]: `Store.current`?

Comment (brendandeere) [inline app/models/solidus_subscriptions/consolidated_installment.rb]: Good catch, Although I dont think I can use `current` with out passing it the `store_key` value which I wont have as this runs in a separate process. 

However I can use the value from `root_order.store` so Subscriptions Orders always get placed for the same store as the root order

**Comment by Murph33** [comment]
lgtm :+1:

**Pull request merged**